olive/engine/engine.py

```diff
@@ -224,6 +235,7 @@ def run(
         output_dir: str = None,
         output_name: str = None,
         evaluate_input_model: bool = True,
+        enable_cloud_cache: "CloudCacheConfig" = None,
```

Comment:
Is this variable named correctly? Should this be `cloud_cache_config`? `enable_cloud_cache` is member of the config. Is the type specifier wrong? Should that be bool?

Reply:
I intentionally to make this config can be both `bool` and `CloudCacheConfig`. If it is a `bool`, it will use all default configurations (one azure storage for all users). If they want to use their own storage, they can use `CloudCacheConfig` to configure the details. Otherwise those who want to use default cloud cache will have to write unnecessary configs.

Reply:
Should that decision be made here? Shouldn't that be part of the config validation so everything downstream from there will get the same input. Making the decision here between `bool` vs. `CloudCacheConfig` means every other system/code will have to also implement similar behavior which is unnecessary. We should have that decision be made once and, in my opinion, the config validation is the right place for it (similar logic exists for `DataConfig` also where the input could be either a `str` or a `DataConfig` object).

Reply:
It is validated by config. Please check line 89 in run/config.py file.

Reply:
> If it is a bool, it will use all default configurations (one azure storage for all users).
> It is validated by config. Please check line 89 in run/config.py file.

I am confused now. In your previous statement you said it could be either/or. But if the `validate_config` is always returns an instance of `CloudCacheConfig` then it cannot be a `bool` as parameter to `Engine`. It will always be an instance of `CloudCacheConfig` or None. - in which case, the variable should be named `cloud_cache_config`.

The config accepting either `bool` or an instance of `CloudCacheConfig` is good. But, the engine should only care for it as an instance of `CloudCacheConfig` (or None) and never as `bool`.

Reply:
Yeah you are right i misunderstood you. For engine it is always `CloudCacheConfig`. I can rename this parameter.